src/foam/java/refinements.js

```diff
@@ -55,6 +55,7 @@ foam.CLASS({
         propName: this.name,
         propType: this.javaType,
         propRequired: this.required,
+        of: this.of,
```

Comment:
Also, most PropertyInfo's don't have an 'of' property, which is where you were getting the 'undefined' from. I think this should only be added to PropertyInfo sub-classes that use 'of'.